sumpy/kernel.py

```diff
@@ -856,6 +857,91 @@ def get_args(self):
                     )
                 ]
 
+
+class LineOfCompressionKernel(ExpressionKernel):
+    """A kernel for the line of compression or dilatation of constant strength
+    along the axis "axis" from zero to negative infinity. This is used for the
+    explicit solution to half-space Elasticity problem. See [1] for details.
+
+    [1]: Mindlin, R.: Force at a Point in the Interior of a Semi-Infinite Solid
+         https://doi.org/10.1063/1.1745385
+    """
+    init_arg_names = ("dim", "axis", "viscosity_mu", "poisson_ratio")
+
+    def __init__(self, dim=3, axis=2, viscosity_mu="mu", poisson_ratio="nu"):
+        r"""
+        :arg axis: axis number defaulting to 2 for the z axis.
+        :arg viscosity_mu: The argument name to use for
+                dynamic viscosity :math:`\mu` when generating functions to
+                evaluate this kernel. Can also be a numeric value.
+        :arg poisson_ratio: The argument name to use for
+                Poisson's ratio :math:`\nu` when generating functions to
+                evaluate this kernel. Can also be a numeric value.
+        """
+        if isinstance(viscosity_mu, str):
+            mu = parse(viscosity_mu)
+        else:
+            mu = viscosity_mu
+        if isinstance(poisson_ratio, str):
+            nu = parse(poisson_ratio)
+        else:
+            nu = poisson_ratio
+
+        if dim == 3:
+            d = make_sym_vector("d", dim)
+            r = pymbolic_real_norm_2(d)
+            # Kelvin solution
+            expr = d[axis] * var("log")(r + d[axis]) - r
+            scaling = (1 - 2*nu)/(4*var("pi")*mu)
+
+        elif dim is None:
+            expr = None
+            scaling = None
```

Comment:
What's this about?

Reply:
I've copied this from other kernels. I have no idea what's it about

Reply:
#55

Reply:
Thanks. Dropped for the new kernel in 90ddf4f